cmd/declaration_extraction/declaration_extraction.go

```diff
@@ -0,0 +1,297 @@
+package declaration_extraction
+
+import (
+	"bufio"
+	"fmt"
+	"os"
+	"regexp"
+	"strings"
+)
+
+type Declaration struct {
+	PackageID   string
+	FileID      string
+	StartOffset int
+	Length      int
+	Name        string
+}
+
+// func isDeclared(decrArr []Declaration, dec Declaration) bool {
+// 	for _, curDec := range decrArr {
+// 		if curDec == dec {
+// 			return true
+// 		}
+// 	}
+// 	return false
+// }
+
+func extractGlbl(source *os.File, fileName string) []Declaration {
+
+	var GlblDec []Declaration
+	var pkgID string
+
+	scanner := bufio.NewScanner(source)
+
+	//Regexs
+	reMultiCommentOpen := regexp.MustCompile(`/\*`)
+	reMultiCommentClose := regexp.MustCompile(`\*/`)
+	reComment := regexp.MustCompile("//")
+
+	reBodyOpen := regexp.MustCompile("{")
+	reBodyClose := regexp.MustCompile("}")
+
+	rePkg := regexp.MustCompile("package")
+	rePkgName := regexp.MustCompile(`(^|[\s])package\s+([_a-zA-Z][_a-zA-Z0-9]*)`)
+
+	reGlbl := regexp.MustCompile("var")
+	reGlblName := regexp.MustCompile(`(^|[\s])var\s([_a-zA-Z][_a-zA-Z0-9]*)`)
+
+	var inBlock int
+	var commentedCode bool
+
+	//Reading code line by line
+	for scanner.Scan() {
+		line := scanner.Bytes()
+
+		commentLoc := reComment.FindIndex(line)
+
+		multiCommentOpenLoc := reMultiCommentOpen.FindIndex(line)
+		multiCommentCloseLoc := reMultiCommentClose.FindIndex(line)
+
+		//Check for commented code
+		if commentedCode && multiCommentCloseLoc != nil {
+			commentedCode = false
+		}
+
+		if commentedCode {
+			continue
+		}
+
+		if multiCommentOpenLoc != nil && !commentedCode && multiCommentCloseLoc != nil {
+			commentedCode = true
+		}
```

Comment:
All functions right now are manually checking comments. We can first get the source code, then remove all the comments, and then the source code that has no comments will be passed to these extractor functions.